src/components/molecules/table-fileds-filter/index.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { useEffect, useRef, useState } from "react"
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu"
+import { sortBy } from "lodash"
+
+import PlusIcon from "../../fundamentals/icons/plus-icon"
+import Button from "../../fundamentals/button"
+import CrossIcon from "../../fundamentals/icons/cross-icon"
+import Checkbox from "../../atoms/checkbox"
+
+/******************** TYPES ********************/
+
+type Field = {
+  id: string
+  short: string
+  label: React.ReactChild | ((args: { isSelected: boolean }) => void)
+}
+
+type ChipProps = Field & {
+  remove: () => void
+}
+
+type TableFieldsFilterProps = {
+  fields: Field[]
+  onChange: (selectedFieldIds: string[]) => void
+}
+
+type FieldsMenuProps = {
+  fields: Field[]
+  onBlur: (ids: string[]) => void
+  selectedFields: string[]
+}
+
+type FieldMenuItemProps = {
+  field: Field
+  checked: boolean
+  onChange: () => void
+}
+
+/******************** COMPONENTS ********************/
+
+/**
+ * Table field chip component.
+ */
+function Chip(props: ChipProps) {
+  const { remove, short } = props
+  return (
+    <div className="rounded rounded-lg h-[32px] flex gap-1 shrink-0 items-center px-3 text-small text-grey-70 border border-gray-70">
+      {short}
+      <CrossIcon
+        className="text-grey-40 cursor-pointer"
+        onClick={remove}
+        size={13}
+      />
+    </div>
+  )
+}
+
+/**
+ * `FieldMenu` item component.
+ */
+function FieldMenuItem(props: FieldMenuItemProps) {
+  const { checked, field, onChange } = props
+  return (
+    <DropdownMenu.Item>
+      <Checkbox
+        checked={checked}
+        className="px-[6px] h-[32px] hover:bg-grey-10 rounded text-small"
+        onChange={onChange}
+        label={
+          typeof field.label === "function"
+            ? field.label({ isSelected: props.checked })
+            : field.label
+        }
+      />
+    </DropdownMenu.Item>
+  )
+}
+
+/******************** CONTAINERS ********************/
+
+/**
+ * The dropdown menu for selecting currently active table fields.
+ */
```

Comment:
nit: I like adding descriptions to the components, but perhaps we should turn off "valid-jsdoc" in the eslint config in that case.

Reply:
@fPolic What do you think of this suggestion?

Reply:
I would prefer if we could kill that particular rule - I think we were planning to do the same thing on the core project?

Reply:
Yes, I agree!